azurerm_container_group/main.tf

```diff
@@ -0,0 +1,90 @@
+provider "azurerm" {
+  version = "=1.42.0"
+}
+
+terraform {
+  backend "azurerm" {}
+}
+
+# Modules Section
+
+module "storage_account" {
+  source = "git::git@github.com:pagopa/io-infrastructure-modules-new.git//azurerm_storage_account?ref=v0.0.5"
+
+  // Global parameters
+  region        = var.region
+  global_prefix = var.global_prefix
+  environment   = var.environment
+
+  // Module parameters
+  name                      = var.storage_account_name
+  resource_group_name       = var.resource_group_name
+  account_kind              = var.storage_account_account_kind
+  account_tier              = var.storage_account_account_tier
+  account_replication_type  = var.storage_account_account_replication_type
+  access_tier               = var.storage_account_access_tier
+}
+
+module "storage_share" {
+  source = "git::git@github.com:pagopa/io-infrastructure-modules-new.git//azurerm_storage_share?ref=v0.0.5"
+  
+  // Global parameters
+  region        = var.region
+  global_prefix = var.global_prefix
+  environment   = var.environment
+
+  // Module parameters
+  name                 = var.storage_share_name
+  storage_account_name = module.storage_account.resource_name
```

Comment:
Dependency between modules is not guaranteed, this means that storage_share do not wait storage_account. Please in "storage_share" use the module provided "module_depends_on" input variable to force dependency.